compiler/src/data/input_data/mod.rs

```diff
@@ -0,0 +1,46 @@
+pub mod compiler_type;
+pub mod header;
+pub mod options;
+pub mod source_code;
+
+
+use std::path::PathBuf;
+
+
+// Simplified structure for the very first step
+use self::compiler_type::CompilerType;
+
+pub struct InputData {
+    pub compiler_type: CompilerType,
+    pub source_code_file_path: PathBuf,  
+    pub compiled_directory_path: PathBuf,
+    pub compiler_options: String,
+}
+
+impl InputData {
```

Comment:
Members are already public, is there a need in constructor?

Reply:
> Members are already public, is there a need in constructor?

Constructor was added a little bit later after struct for the more effective initialization in backend. And now, I suppose we need to make private these pub fields.  Just added corresponding TODO record.